### PR TITLE
Intel SoC FPGA Agilex Development Kit Initial Bring Up

### PR DESCRIPTION
**Pull request by siclim**
This is the initial Zephyr support for Intel SoC FPGA Agilex support.
Agilex has quad-core 64-bit Arm Cortex*-A53.

This patch build Zephyr for Agilex development kit with 256KB SDRAM and
support hello_world sample code. The Zephyr will need to be loaded by
Intel Arm Trusted Firmware (ATF).

Agilex Zephyr boot flow:
	FSBL:ATF BL2(EL3) -> SSBL:ATF BL31(EL3) -> OS:Zephyr(EL2->EL1)

Intel ATF can be downloaded from:
	https://github.com/altera-opensource/arm-trusted-firmware.git

### CODE REVIEW

File: boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk_defconfig
Comment (carlocaione) [inline boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk_defconfig:19]: None of this should actually be needed.
- `CONFIG_XIP` is `n` by default
- `CONFIG_AARCH64_IMAGE_HEADER` is `y` by default now, 
- `CONFIG_SRAM_*` should be defaulted to whatever is in the DTS
- same for `CONFIG_FLASH_*`

File: drivers/clock_control/CMakeLists.txt
Comment (carlocaione) [inline drivers/clock_control/CMakeLists.txt:46]: Why not using `zephyr_sources_ifdef`?

File: drivers/clock_control/clock_agilex_ll.c
Comment (carlocaione) [inline drivers/clock_control/clock_agilex_ll.c:23]: you are returning `uint32_t`

File: drivers/clock_control/clock_agilex_ll.c
Comment (carlocaione) [inline drivers/clock_control/clock_agilex_ll.c:39]: same

File: drivers/clock_control/clock_agilex_ll.c
Comment (carlocaione) [inline drivers/clock_control/clock_agilex_ll.c:61]: This `if` is actually never taken because `count` is at most `10` given the while loop two lines before.

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (carlocaione) [inline soc/arm64/intel_socfpga/agilex/mmu_regions.c:20]: Mapping the whole memory should not be necessary.

File: drivers/clock_control/clock_agilex_ll.c
Comment (carlocaione) [inline drivers/clock_control/clock_agilex_ll.c:90]: What is using this function?

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (carlocaione) [inline soc/arm64/intel_socfpga/agilex/mmu_regions.c:15]: Use `MT_P_RW_U_RW` or similar instead of `MT_RW`

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (carlocaione) [inline soc/arm64/intel_socfpga/agilex/mmu_regions.c:25]: Are you sure you want this to be `MT_SECURE`?

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (carlocaione) [inline soc/arm64/intel_socfpga/agilex/soc.h:33]: Why do you want to define new macros?

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (carlocaione) [inline soc/arm64/intel_socfpga/agilex/soc.h:39]: Use `sys_set_bit` and `sys_clear_bit` instead.

File: soc/arm64/intel_socfpga/common/socfpga_handoff.h
Comment (carlocaione) [inline soc/arm64/intel_socfpga/common/socfpga_handoff.h:95]: Where are this functions defined?

**Comment by carlocaione** [comment]
Board documentation is missing as well.

File: boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk_defconfig
Comment (siclim) [reply boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk_defconfig:19]: Noted. I will remove this when submit version 2 for review.

File: drivers/clock_control/CMakeLists.txt
Comment (siclim) [reply drivers/clock_control/CMakeLists.txt:46]: Noted. I will update to

zephyr_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex_ll.c)
zephyr_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex.c)

File: drivers/clock_control/clock_agilex_ll.c
Comment (siclim) [reply drivers/clock_control/clock_agilex_ll.c:23]: I will update the function to return 'int'. Same for the next function.

File: drivers/clock_control/clock_agilex_ll.c
Comment (siclim) [reply drivers/clock_control/clock_agilex_ll.c:61]: Thanks. I will fix the code to to (count >= 10).

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:20]: Thanks. I have removed this.

File: drivers/clock_control/clock_agilex_ll.c
Comment (siclim) [reply drivers/clock_control/clock_agilex_ll.c:90]: Hi Carlo,
We plan to support and upstream Agilex features by phases.

There will be 2 kinds of Zephyr boot for Agilex:

(1) ATF BL2 -> ATF BL31 -> Zephyr
     - This is what we upstream now, first batch of of Agilex upstream code

(2) Zephyr self-boot, processor directly boot from zephyr.bin
     - We will upstream this flow later. 
     - We will need to call 'z_arm64_el3_plat_init' function to initialize hardware component (example: pinmux, clock)
     - config_clkmgr_handoff will be used later once we enable self-boot flow. And, the clock_agilex_ll.c is shared/migrated from our ATF source code. Thus, we retains the function for future use.

Kindly advise this is acceptable for the mainline code.  Thanks a lot.

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:15]: Noted.

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/soc.h:33]: Hi Carlo,

We will re-use our ATF driver code for Zephyr.  The ATF driver codes are using mmio_X function to write/read. 
We add these macros to map mmio_X to Zephyr compatible function, so that we can re-use most of the code from ATF.

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/soc.h:39]: Hi Carlo,

In our use case, the 'bit' represent mask, our code will set/clear more that 1 bits using this command. 
I will change bit to mask in the macro to confusion.
    mmio_setbits_32(addr, bit) => mmio_setbits_32(addr, mask)

sys_set_bit and sys_clear_bit inline function accept bit offset. It does not fit our function. To centralize bitops in zephyr, is it possible to provide the macro that set/clear bits using mask?
    Example: sys_set_bits (addr, mask)

static ALWAYS_INLINE void sys_set_bit(mem_addr_t addr, unsigned int bit)
{
	uint32_t temp = *(volatile uint32_t *)addr;

	*(volatile uint32_t *)addr = temp | (1 << bit);
}

Thanks,
Siew Chin

File: soc/arm64/intel_socfpga/common/socfpga_handoff.h
Comment (siclim) [reply soc/arm64/intel_socfpga/common/socfpga_handoff.h:95]: These 2 functions are only needed in self-boot flow in socfpga_handoff.c, which I don't upstream in this batch. Sorry for confusion. I will remove these 2 lines.

(2) Zephyr self-boot, processor directly boot from zephyr.bin

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:25]: I have changed it to MT_NS, it doesn't need to be secure.

File: drivers/clock_control/clock_agilex_ll.c
Comment (carlocaione) [reply drivers/clock_control/clock_agilex_ll.c:90]: If you are upstreaming now only the ATF boot path I would expect only to find in this PR code related to that path. So, please remove any unused / not needed code for this PR, you can add that later when upstreaming the other boot flow.

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:25]: Not sure about your use-case but you can think to use `MT_DEFAULT_SECURE_STATE`

File: drivers/clock_control/clock_agilex_ll.c
Comment (siclim) [reply drivers/clock_control/clock_agilex_ll.c:90]: Noted, I will remove all unused code for this phase.

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:25]: Noted, I will change to use MT_DEFAULT_SECURE_STATE. And, will adjust accordingly in future when needed.

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/soc.h:33]: That's not actually correct. You are defining these macros in a soc-specific header file, but you are using the macros in a general driver. Now, let's say (I know this is really unlikely) that someone wants to use your clock agilex driver, in that case they cannot do that (or at least the driver fails to compile) because they have no access to the macros because these are defined in a SoC header file.

So, you have two ways of fixing this:
1. Move the macros out of this header file in a more accessible place so that it can be used together with the drivers using the macros
2. Remove the macros entirely

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/soc.h:39]: > is it possible to provide the macro that set/clear bits using mask?

Feel free to add the new macros you need in `include/arch/common/sys_bitops.h`

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/soc.h:33]: Is it acceptable that I move the mmio macro to the beginning of 'clock_agilex_ll.c' itself? 
This allow us to re-use the existing code.

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/soc.h:39]: Noted.

File: soc/arm64/intel_socfpga/agilex/soc.h
Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/soc.h:33]: Yes, but please add a comment explaining why, otherwise it looks odd :)

Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/soc.h:33]: Hi Carlo,

Noted and thanks.

A side question, I am new to open source upstream process, is that I should close this pull-request now and resubmit a new pull-request after I clean up all codes?

Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/soc.h:33]: Nope, you have to force push your branch with the changes using git `git push --force ....`, that will retrigger the process.

Please also note that besides addressing my comments you have two other issues to address: the compliance and the scancode checks are failing (see the red crosses at the bottom of this PR).

Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/soc.h:33]: Noted and thanks for reminder for the compliance and scancode error. I will check that.

File: drivers/clock_control/clock_agilex_ll.c
Comment (dcpleung) [reply drivers/clock_control/clock_agilex_ll.c:23]: A single line statement for if-block needs curly brackets. Please add.

File: drivers/clock_control/clock_agilex_ll.c
Comment (dcpleung) [inline drivers/clock_control/clock_agilex_ll.c:83]: Same as the comment above about count not reaching 100.

Comment (siclim) [reply drivers/clock_control/clock_agilex_ll.c:83]: Noted. I will update the code in next pull request. Thanks.

**Comment by siclim** [comment]
Hi Reviewers,

Can you please help to review the pull request?

File: boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
Comment (carlocaione) [inline boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts:11]:
```suggestion
#include <intel_socfpga/intel_socfpga_agilex.dtsi>

/ {
```

File: boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
Comment (carlocaione) [inline boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts:25]:
```suggestion
&uart0 {
```

File: drivers/clock_control/clock_agilex.c
Comment (carlocaione) [inline drivers/clock_control/clock_agilex.c:55]: `device_pm_control_nop` is deprecated, use `NULL`

File: dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi
Comment (carlocaione) [inline dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi:12]:
```suggestion
	cpus {
```

File: dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi
Comment (carlocaione) [inline dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi:23]:
```suggestion
	soc {
```

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (carlocaione) [inline soc/arm64/intel_socfpga/agilex/mmu_regions.c:14]: What's this huge hard-coded area?

Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:14]: This is the peripheral register map range for all peripheral in Agilex. We enable for OS access for initial bring.

Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:14]: Uhm, I see two problems here:
1. You are missing at least two device regions here: UART and CLOCK that have entries in the DT (you only have GIC).
2. Why do you need to map the whole peripheral range here? Do you have some kind of application running that needs that?

Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:14]: UART and CLOCK register are within this huge peripheral register range, we just enable for all during bring up without specific application.

Agreed that this is a problem.

I will do a clean up to only enable for UART and CLOCK, which means only enable the peripheral that we enabled in the code.

File: dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi
Comment (siclim) [reply dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi:12]: Okay, update.

File: dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi
Comment (siclim) [reply dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi:23]: Okay, updated.

File: boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
Comment (siclim) [reply boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts:11]: Okay. Updated.

File: boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
Comment (siclim) [reply boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts:25]: Okay. Updated.

File: drivers/clock_control/clock_agilex.c
Comment (siclim) [reply drivers/clock_control/clock_agilex.c:55]: Noted, updated.

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:14]: Remove huge area of PERIPERAL from MMU table, only include:
(1) SYSTEM_MANAGER and CLOCK register map for clock driver
(2) UART0 register map for uart driver

**Comment by siclim** [comment]
I have updated the patches based on review and force pushed to the branch. Kindly help to review. Thanks a lot.

File: soc/arm64/intel_socfpga/agilex/mmu_regions.c
Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:16]: syscon-like device? I am not sure how syscon-like device is, so i can't confirm.

Agilex System Manager register:
https://www.intel.com/content/www/us/en/programmable/hps/agilex/hps.html#jwd1561687644789.html

System Manager is a block inside HPS (Hard processer system) in Agilex SoCFPGA, it provide some settings for the periperal interface within HPS. 

Clock driver will need to read the external clock freq from boot_scratch_cold1 and boot_scratch_cold2 registers in system manager clock. cold1 and cold2 register is defined for custom used, in our implementation, we  use these 2 register to stores the external system clock system from user settings. So we need to access it in clock driver to calculate the frequently.

Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:16]: **syscon** comes from the linux world https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/mfd/syscon.yaml and it represents  _a register region containing a set of miscellaneous registers. The registers are not cohesive enough to represent as any specific type of device_ and unfortunately this is exactly what it looks like to me.

Now, the problem is that you are defining here a device region without a backing device or driver and that should be always avoided but on the other hand Zephyr is not shipping any syscon driver for that.

I cannot ask you to write a syscon driver (even though it would be really easy to do) but please, at least use a couple of defines for the reg / size values and add a comment explaining what this region is.

Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:16]: I have taka a look at syscon.yaml, and, yes that this is what I need. The system manager register will be access by different driver or feature in future, we just need a way to access the register directly.

I agree with you that it is cleaner to create a "syscon" device tree node for these register range. I can add this new device tree node element and driver. If there any concern if I proceed to add it?

Can we use back "syscon" as the name for this element? I expect this element only need to support 2 properties: "compatible" and "reg". We can add more properties in future when needed.

Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:16]: I will proceed to add a driver call system_mgr for our device.

Comment (carlocaione) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:16]: I'm writing a syscon driver right now, I will push it tomorrow.

Comment (siclim) [reply soc/arm64/intel_socfpga/agilex/mmu_regions.c:16]: Noted, thanks. I will wait for your change. :)

**Comment by cfriedt** [comment]
@siclim - please resolve conflicts and re-request reviews

**Comment by siclim** [comment]
Hi, I have dependency on https://github.com/zephyrproject-rtos/zephyr/pull/35903 and I am still waiting for it to be merged into main.

**Comment by cfriedt** [comment]
> Hi, I have dependency on #35903 and I am still waiting for it to be merged into main.

@siclim - you should be good to go now.

**Comment by siclim** [comment]
Noted and thanks.

**Comment by siclim** [comment]
Sorry for my late update. Can you please help to remove the 'stale' lable and help to review the update?

I have updated the to use 'syscon' device as per request to specify the register map for SYSTEM MANAGER in Agilex. And, MMU will get the start address and size from the sysmgr device tree.

	MMU_REGION_FLAT_ENTRY("SYSTEM_MANAGER",
			      DT_REG_ADDR(DT_NODELABEL(sysmgr)),
			      DT_REG_SIZE(DT_NODELABEL(sysmgr)),
			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_DEFAULT_SECURE_STATE),

File: include/arch/common/sys_bitops.h
Comment (nashif) [inline include/arch/common/sys_bitops.h:45]: please document the new APIs in include/sys/sys_io.h

**Review by nashif — CHANGES_REQUESTED** [review]
You need to add intel_socfpga_agilex_socdk.yaml file into the board directory, this will allow integration with the test runner. Once you have the yaml file, try running:

`twister -p intel_socfpga_agilex_socdk`

and make sure everything builds correctly.

File: include/arch/common/sys_bitops.h
Comment (siclim) [reply include/arch/common/sys_bitops.h:45]: I have added intel_socfpga_agilex_socdk.yaml and run twister but there are "Cmake build failure". Is it expected?

swiplab@PG15SWIPLAB1115:~/siclim/zephyrproject/zephyr$ ./scripts/twister -p intel_socfpga_agilex_socdk
ZEPHYR_BASE unset, using "/home/swiplab/siclim/zephyrproject/zephyr"
Install the anytree module to use the --test-tree option
Install tabulate python module with pip to use --device-testing option.
Install pyserial python module with pip to use --device-testing option.
Install tabulate python module with pip to use --device-testing option.
Install psutil python module with pip to run in Qemu.
can't import pykwalify; won't validate YAML (No module named 'pykwalify')
INFO    - Zephyr version: v2.7.0-rc2-144-g82c1c10d85d7
INFO    - JOBS: 12
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 1466 test scenarios (1406 configurations) selected, 1017 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    3/ 389   0%  skipped: 1020, failed:    0ERROR   - Cmake build failure: /home/swiplab/siclim/zephyrproject/zephyr/samples/modules/nanopb for intel_socfpga_agilex_socdk

ERROR   - intel_socfpga_agilex_socdk samples/modules/nanopb/sample.modules.nanopb       FAILED: Cmake build failure
ERROR   - see: /home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/samples/modules/nanopb/sample.modules.nanopb/build.log

Comment (nashif) [reply include/arch/common/sys_bitops.h:45]: Not, this is not expected. you need to install the python dependencies, here pykwalify, look at the getting started guide

Comment (siclim) [reply include/arch/common/sys_bitops.h:45]: Thanks, it can be built after install all the dependencies. There are 2 build error:

twister.log
----------
2021-10-02 04:36:33,849 - twister - ERROR - intel_socfpga_agilex_socdk tests/posix/fs/portability.posix.fs                [31mFAILED[39m: Build failure
2021-10-02 04:36:33,850 - twister - ERROR - see: [33m/home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/tests/posix/fs/portability.posix.fs/build.log[39m
2021-10-02 04:36:33,925 - twister - ERROR - intel_socfpga_agilex_socdk tests/posix/fs/portability.posix.fs.newlib         [31mFAILED[39m: Build failure
2021-10-02 04:36:33,925 - twister - ERROR - see: [33m/home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/tests/posix/fs/portability.posix.fs.newlib/build.log[39m

build.log
----------
-- Found BOARD.dts: /home/swiplab/siclim/zephyrproject/zephyr/boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
intel_socfpga_agilex_socdk.dts.pre.tmp:28.21-39.5: Warning (simple_bus_reg): /soc/timer: missing or empty reg/ranges property
....
zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/build.make:103: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj' failed
make[2]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj] Error 1
CMakeFiles/Makefile2:2796: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all' failed
make[1]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2


I didn't see any error in the build.log, can you please advise is  warning 'missing or empty reg/ranges property' caused the build failed? I have tried to refer to other dts in Zephyr, I do see any difference on soc/timer when compare with others.

Comment (nashif) [reply include/arch/common/sys_bitops.h:45]: good, you need to resolve those issues and make sure they are not coming with your change.

Comment (siclim) [reply include/arch/common/sys_bitops.h:45]: I have fixed all warning caused by my changes. 

The twister is failed with the following error which is not caused by my changes. Am i good to resubmit my patches for review?

zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/build.make:103: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj' failed
make[2]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj] Error 1
CMakeFiles/Makefile2:2798: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all' failed
make[1]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all] Error 2
Makefile:90: recipe for target 'all' failed

Comment (siclim) [reply include/arch/common/sys_bitops.h:45]: I have resubmitted the patches. All warnings coming from my changes are fixed.

**Comment by siclim** [comment]
4 updates:

1. Document newly added sys_set_bits and set_clear_bits API in include/sys/sys_io.h
2. Change zephyr_sources_ifdef to zephyr_library_sources_ifdef in drivers/clock_control/CMakeLists.txt to fix the warning reported in twister.
3. Remove soc/ layer in dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi file to fix 'Warning (simple_bus_reg): /soc/timer: missing or empty reg/ranges property'
4. Add boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.yaml file to enable unit test run on intel socfpga.

**Comment by nashif** [comment]
@carlocaione can you please give this another look?

**Review by carlocaione — APPROVED** [review]
It looks sane to me but @siclim please remove the `Update:` lines from the commit messages (in theory those should not appear in the commit message after merge but not sure how github handles that).

**Comment by siclim** [comment]
Hi @anasnashif , 

All checks have passed, can you please help to review again? Thanks a lot.

**Review by nashif — APPROVED** [review]

**Pull request merged**
